src/monitors/memberProfiles.js

```diff
@@ -16,56 +16,60 @@ module.exports = class extends Monitor {
+        const newLevel = Math.floor(0.2 * Math.sqrt(newXP));
+        await msg.member.settings.update([["xp", newXP], ["level", newLevel]]);
 
         timeout.add(`${msg.author.id}-${msg.guild.id}`);
         setTimeout(() => timeout.delete(`${msg.author.id}-${msg.guild.id}`), 45000);
 
-        // Generate Level Up Images on Level Up
-        if (oldLvl !== newLvl) {
-            if (msg.guild.settings.levelup) {
-                if (msg.guild.settings.leveltype !== "guild") return;
-                if (!msg.channel.postable) return;
-                const bgName = msg.author.settings.profilebg;
-                const bgImg = await fs.readFile(`../assets/profiles/bg/${bgName}.png`);
-                const avatar = await get(msg.author.displayAvatarURL({ format: "png", size: 128 })).then(res => res.body).catch(e => {
-                    Error.captureStackTrace(e);
-                    return e;
-                });
-                const img = await new Canvas(100, 100)
-                    .addImage(bgImg, 0, 0, 530, 530)
-                    .addImage(avatar, 22, 22, 57, 57)
-                    .toBufferAsync();
-                msg.sendMessage(`🆙 | **${msg.author.tag}** has leveled up to **Level ${newLvl}** in **${msg.guild.name}**`, { files: [{ attachment: img, name: `${msg.author.id}.png` }] });
-            }
-            if (msg.guild.settings.levelroles.enabled) {
-                const { roles } = msg.guild.settings.levelroles;
-                if (roles.length === 0) return;
-                const levelRole = roles.find(r => r.lvl === newLvl);
-                if (!levelRole) return;
-                const role = msg.guild.roles.find(r => r.id === levelRole);
-                if (!role) return;
-                const myRole = msg.guild.me.roles.find(r => r.managed);
-                if (role.position > myRole.positon) return;
-                if (msg.member.roles.has(role)) return;
-                await msg.member.roles.add(role, "Level Based Role").catch(() => null);
-            }
-        }
+        if (oldLevel === newLevel) return;
+
+        const { settings } = msg.guild;
+
+        if (settings.levelup && settings.leveltype === "guild") await this.handleLevelup(msg);
+        if (settings.levelroles.enabled && settings.levelroles.roles.length) await this.leveledroles(msg);
+    }
+
+    async handleLevelup(msg) {
+        if (!msg.channel.postable) return;
+
+        const image = await this.generateLevelUpImage(msg.author.settings.profilebg, msg.author.displayAvatarURL({ format: "png", size: 128 }));
+        return msg.sendMessage(`🆙 | **${msg.author.tag}** has leveled up to **Level ${msg.member.settings.level}** in **${msg.guild.name}**`, { files: [{ attachment: image, name: `${msg.author.id}.png` }] });
+    }
+
+    async leveledroles(msg) {
+        const levelRole = msg.guild.settings.levelroles.roles.find(r => r.lvl === msg.member.settings.level);
```

Comment:
Make it so it adds all the roles prior to their current level even if the member is passed that certain level in the guild.